Makefile

```diff
@@ -18,12 +18,34 @@ check:
 bin/kube-aws-approver: $(GOFILES)
 	@go build $(GOFLAGS) -o $(ROOT_DIR)/bin/kube-aws-approver github.com/coreos/kubecsr/cmd/kube-aws-approver
 
+bin/kube-signer-server: $(GOFILES)
+	@go build $(GOFLAGS) -o $(ROOT_DIR)/bin/kube-signer-server github.com/coreos/kubecsr/cmd/kube-signer-server
+
+bin/kube-client-agent: $(GOFILES)
+	@go build $(GOFLAGS) -o $(ROOT_DIR)/bin/kube-client-agent github.com/coreos/kubecsr/cmd/kube-client-agent
+
 image/kube-aws-approver:
-	@docker build -t quay.io/coreos/kube-aws-approver:$(IMAGE_TAG) -f $(ROOT_DIR)/Dockerfile.kube-aws-approver .
+	@docker build -t quay.io/coreos/kube-aws-approver:$(IMAGE_TAG) -f $(ROOT_DIR)/dockerfiles/Dockerfile.kube-aws-approver .
 
 push/kube-aws-approver: image/kube-aws-approver
 	@docker push quay.io/coreos/kube-aws-approver:$(IMAGE_TAG)
 
+image/kube-signer-server:
+	@docker build -t quay.io/coreos/kube-signer-server:$(IMAGE_TAG) -f $(ROOT_DIR)/dockerfiles/Dockerfile.kube-signer-server .
+
+push/kube-signer-server: image/kube-signer-server
+	@docker push quay.io/coreos/kube-signer-server:$(IMAGE_TAG)
+
+image/kube-client-agent:
+	@docker build -t quay.io/coreos/kube-client-agent:$(IMAGE_TAG) -f $(ROOT_DIR)/dockerfiles/Dockerfile.kube-client-agent .
+
+push/kube-client-agent: image/kube-client-agent
+	@docker push quay.io/coreos/kube-client-agent:$(IMAGE_TAG)
+
+test:
```

Comment:
`make check` already runs `go test`
also just `go test -v $(shell go list ./... | grep -v '/e2e')` should be sufficient here.

Reply:
This is take from [dex Makefile](https://github.com/coreos/dex/blob/master/Makefile).

I added this so you can run just the unit tests separately from everything else running in make check.

whats the reasoning for using  grep -v '/e2e' vs grep -v '/vendor/'

Reply:
I think go default tool-chain commands skips `vendor/` dir when running `./...`